src/storage/src/table/cell_based_table.rs

```diff
@@ -197,58 +198,74 @@ impl<S: StateStore> CellBasedTable<S> {
 
     async fn batch_write_rows_inner<const WITH_VALUE_META: bool>(
         &mut self,
-        buffer: BTreeMap<Row, RowOp>,
+        buffer: BTreeMap<Vec<u8>, RowOp>,
         epoch: u64,
     ) -> StorageResult<()> {
         // stateful executors need to compute vnode.
         let mut batch = self.keyspace.state_store().start_write_batch();
         let mut local = batch.prefixify(&self.keyspace);
-        let ordered_row_serializer = self.pk_serializer.as_ref().unwrap();
+        // let ordered_row_serializer = self.pk_serializer.as_ref().unwrap();
         let hash_builder = CRC32FastBuilder {};
         for (pk, row_op) in buffer {
-            let arrange_key_buf = serialize_pk(&pk, ordered_row_serializer).map_err(err)?;
-
-            let value_meta = if WITH_VALUE_META {
-                // If value meta is computed here, then the cell based table is guaranteed to have
-                // distribution keys. Also, it is guaranteed that distribution key indices will
-                // not exceed the length of pk. So we simply do unwrap here.
-                let vnode = pk
-                    .hash_by_indices(self.dist_key_indices.as_ref().unwrap(), &hash_builder)
-                    .unwrap()
-                    .to_vnode();
-                ValueMeta::with_vnode(vnode)
-            } else {
-                ValueMeta::default()
-            };
+            // If value meta is computed here, then the cell based table is guaranteed to have
+            // distribution keys. Also, it is guaranteed that distribution key indices will
+            // not exceed the length of pk. So we simply do unwrap here.
 
             match row_op {
                 RowOp::Insert(row) => {
+                    let value_meta = if WITH_VALUE_META {
+                        let vnode = row
+                            .hash_by_indices(self.dist_key_indices.as_ref().unwrap(), &hash_builder)
```

Comment:
Here `dist_key_indices` is the indices of dist key in pk, but not the whole value. For example, row is [v1, v2, v3], pk is  [v2, v3], dist key is [v2], then `dist_key_indices` here will be [0] instead of [1]. You may have to remove the logic of index transformation in the constructor of executor. Currently (maybe) only MV has done this transformation of index.

Reply:
Thanks for your remind,  I will fix it later.